src/indicators/algorithms/exponentialMovingAverage.js

```diff
@@ -0,0 +1,49 @@
+(function(d3, fc) {
+    'use strict';
+
+    fc.indicators.algorithms.exponentialMovingAverage = function() {
+
+        var days = 9,
+            value = fc.utilities.fn.identity;
+
+        function undefinedArrayOfLength(length) {
+            return Array.apply(undefined, new Array(length));
+        }
+
+        var exponentialMovingAverage = function(data) {
```

Comment:
Yes, it can be re-written using `data.map`: https://github.com/ColinEberhardt/d3-financial-components/commit/a772faa0ae34f52fa981239c7749f10830ceb1ae - the implementation style I came up with was as a result of test-first development ;-)

I'm not sure I like the `map` implementations because both have side-effects as a result of the mapping function. However, your point about the array length is a good one.